lib/xcodeproj/scheme.rb

```diff
@@ -11,61 +22,156 @@ class XCScheme
-      @scheme.attributes['LastUpgradeVersion'] = Constants::LAST_UPGRADE_CHECK
-      @scheme.attributes['version'] = '1.3'
-
-      @build_action = @scheme.add_element 'BuildAction'
-      @build_action.attributes['parallelizeBuildables'] = 'YES'
-      @build_action.attributes['buildImplicitDependencies'] = 'YES'
-      @build_action_entries = nil
-
-      @test_action = @scheme.add_element 'TestAction'
-      @test_action.attributes['selectedDebuggerIdentifier'] = 'Xcode.DebuggerFoundation.Debugger.LLDB'
-      @test_action.attributes['selectedLauncherIdentifier'] = 'Xcode.DebuggerFoundation.Launcher.LLDB'
-      @test_action.attributes['shouldUseLaunchSchemeArgsEnv'] = 'YES'
-      @test_action.attributes['buildConfiguration'] = 'Debug'
-
-      @test_action_additional_options = @test_action.add_element 'AdditionalOptions'
-
-      @testables = @test_action.add_element 'Testables'
-
-      @launch_action = @scheme.add_element 'LaunchAction'
-      @launch_action.attributes['selectedDebuggerIdentifier'] = 'Xcode.DebuggerFoundation.Debugger.LLDB'
-      @launch_action.attributes['selectedLauncherIdentifier'] = 'Xcode.DebuggerFoundation.Launcher.LLDB'
-      @launch_action.attributes['launchStyle'] = '0'
-      @launch_action.attributes['useCustomWorkingDirectory'] = 'NO'
-      @launch_action.attributes['buildConfiguration'] = 'Debug'
-      @launch_action.attributes['ignoresPersistentStateOnLaunch'] = 'NO'
-      @launch_action.attributes['debugDocumentVersioning'] = 'YES'
-      @launch_action.attributes['debugServiceExtension'] = 'internal'
-      @launch_action.attributes['allowLocationSimulation'] = 'YES'
-      @launch_action.add_element('AdditionalOptions')
-
-      @profile_action = @scheme.add_element 'ProfileAction'
-      @profile_action.attributes['shouldUseLaunchSchemeArgsEnv'] = 'YES'
-      @profile_action.attributes['savedToolIdentifier'] = ''
-      @profile_action.attributes['useCustomWorkingDirectory'] = 'NO'
-      @profile_action.attributes['buildConfiguration'] = 'Release'
-      @profile_action.attributes['debugDocumentVersioning'] = 'YES'
-
-      analyze_action = @scheme.add_element 'AnalyzeAction'
-      analyze_action.attributes['buildConfiguration'] = 'Debug'
-
-      archive_action = @scheme.add_element 'ArchiveAction'
-      archive_action.attributes['buildConfiguration'] = 'Release'
-      archive_action.attributes['revealArchiveInOrganizer'] = 'YES'
+    # Create a XCScheme either from scratch or using an existing file
+    #
+    # @param [String] file_path
+    #        The path of the existing .xcscheme file. If nil will create an empty scheme
+    #
+    def initialize(file_path = nil)
+      if file_path
+        @doc = REXML::Document.new(File.new(file_path))
+        @doc.context[:attribute_quote] = :quote
+
+        @scheme = @doc.elements['Scheme']
+        raise Informative, 'Unsupported scheme version' unless @scheme.attributes['version'] == Xcodeproj::Constants::XCSCHEME_FORMAT_VERSION
```

Comment:
Pretty sure we just shouldn't check the version, but I'll let @neonichu weigh in on this one